contrib/seccomp-notify-plugin-example-rust/src/seccompwatcher.rs

```diff
@@ -0,0 +1,244 @@
+                                                        _seccomp_fd: i32,
+                                                        shandled: *mut i32) -> i32
+{
+    let req = unsafe {&mut *sreq};
+    let resp = unsafe {&mut *sresp};
+    let handled = unsafe {&mut *shandled};
+
+    if req.data.nr == nc::SYS_MKNOD as i32 {
+        *handled = RUN_OCI_SECCOMP_NOTIFY_HANDLE_SEND_RESPONSE;
+
+        match handle_mknod_request(req) {
+            Ok(_) => {
+                resp.error = 0;
+            }
+            Err(errno) => {
+                if errno == nc::EPERM {
+                    *handled = RUN_OCI_SECCOMP_NOTIFY_HANDLE_SEND_RESPONSE_AND_CONTINUE;
+                } else {
+                    resp.error = -errno;
+                }
+            }
+        }
+
+        resp.id = req.id;
+        resp.val = 0;
+        return 0;
+    }
+    if req.data.nr == nc::SYS_MKNODAT as i32 {
+        *handled = RUN_OCI_SECCOMP_NOTIFY_HANDLE_SEND_RESPONSE;
+
+        match handle_mknodat_request(req) {
+            Ok(_) => {
+                resp.error = 0;
+            }
+            Err(errno) => {
+                if errno == nc::EPERM {
+                    *handled = RUN_OCI_SECCOMP_NOTIFY_HANDLE_SEND_RESPONSE_AND_CONTINUE;
+                } else {
+                    resp.error = -errno;
+                }
+            }
+        }
+
+        resp.id = req.id;
+        resp.val = 0;
+        return 0;
+    }
+
+    *handled = RUN_OCI_SECCOMP_NOTIFY_HANDLE_NOT_HANDLED;
+ 0
+}
+
+fn get_string_from_buffer(fname_buffer: Vec<u8>) -> Result<String, nc::Errno> {
+    let len = match fname_buffer.iter().position(|&x| x == 0) {
+        Some(s) => {
+            s
+        }
+        None => {
+            return Err(nc::EINVAL);
+        }
+
+    };
+    let f = match CString::new(&fname_buffer[0..len]) {
+        Ok(s) => {
+            s
+        }
+        Err(_) => {
+            return Err(nc::EINVAL);
+        }
+    };
+    let filename = match f.into_string() {
+        Ok(s) => {
+            s
+        }
+        Err(_) => {
+            return Err(nc::EINVAL);
+        }
+    };
+    Ok(filename)
+}
+
+fn handle_mknodat_request(req: &mut nc::seccomp_notif_t) -> Result<u32, nc::Errno> {
+    return handle_mknod_internal(0, 1, 2, 3, req)
+}
+
+fn handle_mknod_request(req: &mut nc::seccomp_notif_t) -> Result<u32, nc::Errno> {
+    return handle_mknod_internal(-1, 0, 1, 2, req)
+
+}
+
+fn get_file_from_fd(fd: std::os::unix::io::RawFd) -> std::fs::File {
+    let r = unsafe {std::fs::File::from_raw_fd(fd)};
+    r
+}
+
+fn handle_mknod_internal(dirfd: isize, path_arg: usize, mode_arg: usize, dev_arg: usize, req: &mut nc::seccomp_notif_t) -> Result<u32, nc::Errno> {
+    let mut fname_buffer = vec![0; nc::types::PATH_MAX as usize];
+
+    let fd = get_file_from_fd(nc::open(&format!("/proc/{}/mem", req.pid), nc::O_RDONLY, 0)?);
+
+    nc::pread64(fd.as_raw_fd(), &mut fname_buffer, req.data.args[path_arg] as isize)?;
```

Comment:
For example `pread64` is already exposed in the standard library as [read_at](https://doc.rust-lang.org/std/os/unix/fs/trait.FileExt.html#tymethod.read_at).